pkg/util/github/workflow.go

```diff
@@ -141,3 +141,28 @@ func (c *Client) VerifyWorkflows(workflows []*Workflow) (map[string]error, error
 	}
 	return retMap, nil
 }
+
+func (c *Client) GetWorkflowState() (map[string]interface{}, error) {
```

Comment:
Maybe we don't need to put it under pkg/util, because the return value (which is a state) is only related to some plugins, not all.

Or, we can split this function into two, one gets the workflow, the other creates the state based on the workflow. Getting the workflow is more generic thus fits in the pkg/util; but building state isn't.